stan/math/prim/core/complex_base.hpp

```diff
@@ -0,0 +1,275 @@
+   * @param[in] other a scalar value of matching type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator+=(const X& other) {
+    re_ += other;
+    return derived_complex_ref();
+  }
+
+  /**
+   * Adds other to this.
+   *
+   * @tparam value type of complex argument (assignable to value type)
+   * @param[in] other a complex value of compatible type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator+=(const std::complex<X>& other) {
+    re_ += other.real();
+    im_ += other.imag();
+    return derived_complex_ref();
+  }
+
+  /**
+   * Subtracts other from this.
+   *
+   * @tparam X type of scalar argument (assignable to vlaue type)
+   * @param[in] other a scalar value of matching type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator-=(const X& other) {
+    re_ -= other;
+    return derived_complex_ref();
+  }
+
+  /**
+   * Subtracts other from this.
+   *
+   * @tparam X value type of complex argument (assignable to value type)
+   * @param[in] other a complex value of compatible type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator-=(const std::complex<X>& other) {
+    re_ -= other.real();
+    im_ -= other.imag();
+    return derived_complex_ref();
+  }
+
+  /**
+   * Multiplies this by other.
+   *
+   * @tparam X type of scalar argument (assignable to value type)
+   * @param[in] other a scalar value of matching type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator*=(const X& other) {
+    re_ *= other;
+    im_ *= other;
+    return derived_complex_ref();
+  }
+
+  /**
+   * Multiplies this by other.
+   *
+   * @tparam X value type of complex argument (assignable to value type)
+   * @param[in] other a complex value of compatible type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator*=(const std::complex<X>& other) {
+    value_type re_temp = re_ * other.real() - im_ * other.imag();
+    im_ = re_ * other.imag() + other.real() * im_;
+    re_ = re_temp;
+    return derived_complex_ref();
+  }
+
+  /**
+   * Divides this by other.
+   *
+   * @tparam X type of scalar argument (assignable to value type)
+   * @param[in] other a scalar value of matching type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator/=(const X& other) {
+    re_ /= other;
+    im_ /= other;
+    return derived_complex_ref();
+  }
+
+  /**
+   * Divides this by other.
+   *
+   * @tparam X value type of complex argument (assignable to value type)
+   * @param[in] other a complex value of compatible type
+   * @return this complex number
+   */
+  template <typename X>
+  complex_type& operator/=(const std::complex<X>& other) {
+    using stan::math::square;
+    value_type sum_sq_im = square(other.real()) + square(other.imag());
```

Comment:
Can/should these be `auto`?

Reply:
What's the feeling about `auto` when there's a simple type we can use?  I find the explicit types simpler if they're small.  I'm not bothered either way, so let me know.  Our existing code is all over the place on this issue.

Reply:
I personally like `auto` so that type deduction just rides through instead of performing an implicit cast if there is a mismatch. I'm fine either way here